content/chat/typing.textile

```diff
@@ -5,3 +5,99 @@ product: chat
+
+h3(#enter). Enter
+
+The @enter@ event is intentionally initiated by the client on behalf of the chat participant, allowing them to signify actions such as marking themselves as online upon logging into the platform. For instance, when dealing with a chat participant represented by a user object featuring a @status@ field, the @enter@ event can be triggered using the provided code snippet below. This code serves the purpose of setting the status of the chat participant to @online@, indicating their entry to the conversation.
+
+```[realtime_javascript]
+    // Enter the chat participant into the presence set
+    channel.presence.enter((err) => {
+      // Custom functionality to update UI for chat participant that they are now online.
+    });
+```
+
+h3(#update). Update
+
+The chat participant status can be used as the indicator of whether they're currently typing or not, allowing other chat participants to see a visual cue. The status field is a text based field, so the value can be whichever text value of your choosing.
+
+```[realtime_javascript]
+    // Subscribe to any presence updates for the current presence set
+    channel.presence.subscribe('update', (memberUpdate) => {
+      // Update current array of users typing to display the one for memberUpdate as typing.
+    });
+```
+
+Depending on the UI, it is recommended to display a limited number of the names of chat participants typing. If that number is exceeded, it's common practice to instead change the visual indicator to a string such as "People are typing…".
+
+h3(#example). Demo example
+
+Below is an example of each part required for the typing indicator to be implemented. Once the client has an authToken with a clientId associated with it, you join the predefined channel and enter the presence set for that channel.
+
+The application needs to subscribe to any updates of that presence set and would then update the visual cues displaying which chat participants are currently typing.
+
+For the chat participant, an event listener waits for a keypress before publishing an update to the presence set, updating this chat participant's status with a string indicating they're typing. A timer is set to expire in five seconds to reset this status if there has been no further keys pressed in that time.
+
+```[realtime_javascript]
+    let isTyping = false;
+    let typingInterval;
+    let usersTyping = [];
+
+    var client = new Ably.Realtime(authToken);
+    var channel = client.channels.get('CHANNEL_NAME');
+
+    // Subscribe to any presence updates for the current presence set
+    channel.presence.subscribe('update', (memberUpdate) => {
```

Comment:
We should probably show how the "is typing" and "is not typing" links up here - otherwise it might give the impression that it's some magical thing that automatically handles typing whereas in reality its just a custom string in this demo app :)